packages/app-folders/src/components/Tree/List/index.tsx

```diff
@@ -0,0 +1,214 @@
+import React, { useState } from "react";
+
+import {
+    DropOptions,
+    getBackendOptions,
+    InitialOpen,
+    MultiBackend,
+    NodeModel,
+    Tree
+} from "@minoru/react-dnd-treeview";
+import { useSnackbar } from "@webiny/app-admin";
+import { DndProvider } from "react-dnd";
+import useDeepCompareEffect from "use-deep-compare-effect";
+
+import { FolderDialogDelete, FolderDialogUpdate } from "~/components";
+import { Node } from "~/components/Tree/Node";
+import { NodePreview } from "~/components/Tree/NodePreview";
+import { Placeholder } from "~/components/Tree/Placeholder";
+
+import { useFolders } from "~/hooks";
+
+import { DndItemData, FolderItem } from "~/types";
+
+const ROOT_ID = "root";
+
+const createTreeData = (
+    folders: FolderItem[] = [],
+    focusedNodeId?: string
+): NodeModel<DndItemData>[] => {
+    return folders.map(item => {
+        const { id, parentId, name, slug, type, createdOn, createdBy } = item;
+
+        return {
+            id,
+            parent: parentId || ROOT_ID,
+            text: name,
+            droppable: true,
+            data: {
+                id,
+                name,
+                slug,
+                parentId,
+                type,
+                createdOn,
+                createdBy,
+                isFocused: focusedNodeId === id
+            }
+        };
+    });
+};
+
+const createInitialOpenList = (
```

Comment:
Can we move these outside of this file, into sibling files or a sibling `utils.ts` file?